### DISCUSSION TITLE
What compiler option "ignore_optional_problems" classpath settings maps to

### DISCUSSION THREAD

**Discussion by laeubi** [eclipse-jdt/eclipse.jdt.core#2239] [Q&A]
A classpath in JDT project can have the following entry:

```
<classpathentry kind="src" path=src-gen/">
        <attributes>
                <attribute name="ignore_optional_problems" value="true"/>
        </attributes>
</classpathentry>
```

The help page only says:
> **Ignore optional compile problems**: When set to 'Yes', all optional problems will not be reported by the compiler.

I like to [replicate the behavior in a Tycho](https://github.com/eclipse-tycho/tycho/issues/3719) build and wonder what are "optional problems", does it mean simply ignore all warnings like `<compilerArgument>-nowarn:[${basedir}/src-gen]</compilerArgument>`?

**Comment by iloveeclipse** [comment]
See https://bugs.eclipse.org/bugs/show_bug.cgi?id=220928 and call hierarchy of `org.eclipse.jdt.internal.core.ClasspathEntry.ignoreOptionalProblems()`.

**Comment by stephan-herrmann** [comment]
> what are "optional problems"?

Problems with configurable severity, which explicitly excludes all errors mandated by JLS. Also things like build path errors can by their very nature not be ignored.
